torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -49,5 +49,18 @@ Tensor CosineEmbeddingLossImpl::forward(
   return F::cosine_embedding_loss(input1, input2, target, options);
 }
 
+// ============================================================================
+
+SmoothL1LossImpl::SmoothL1LossImpl(
+    const torch::nn::SmoothL1LossOptions& options_) : options(options_) {}
+
+void SmoothL1LossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::SmoothL1Loss(reduction=" << options.reduction() << ")";
```

Comment:
It would be awesome to add a test for this as well :D

Reply:
As part of the pretty_print test, we can also test that `SmoothL1Loss(/*reduction=*/Reduction::Mean)` (i.e. the implicit constructor of `SmoothL1LossOptions`) works :D

Reply:
Hi @yf225, first of all, thanks for the reviews!
I had deliberately left the `pretty_print` tests out, since there were no such tests for the `L1Loss`, which has the same and unique parameter (`reduction`) as `SmoothL1Loss`, `MultiMarginLoss` and `SoftMarginLoss`. I guessed this was due to #25883, where you pointed out 

> there are still a lot of preparation work that needs to be done (e.g. fixing how we handle enums in C++ API #15149).

and these losses use the `Reduction::Reduction` enum... Should I use a `switch` statement like the one below?

```cpp
void SmoothL1LossImpl::pretty_print(std::ostream& stream) const {
  stream << "torch::nn::SmoothL1Loss(reduction=";
  switch(options.reduction()) {
    case Reduction::None:
      stream << "Reduction::None";
    break;
    case Reduction::Mean:
      stream << "Reduction::Mean";
    break;
    case Reduction::Sum:
      stream << "Reduction::Sum";
    break;
  }
  stream << ")";
}
```